gym-unity/gym_unity/envs/__init__.py

```diff
@@ -180,25 +185,23 @@ def step(self, action: List[Any]) -> GymStepResult:
             return self._single_step(decision_step)
 
     def _single_step(self, info: Union[DecisionSteps, TerminalSteps]) -> GymStepResult:
-        if self.use_visual:
+        if self._allow_multiple_obs:
```

Comment:
I think you can DRY this up a bit similar to the observation space setup. optional though.